tests/test_identify.py

```diff
@@ -1,26 +1,160 @@
+        for x in accepting:
+            assert my_dfa.label(x)
+
+        for x in rejecting:
+            assert not my_dfa.label(x)
+
+        # check that returned DFA is minimal (check bypassed if problem.size = 0)
+        assert (problem.size == 0) or len(my_dfa.states()) == problem.size
+        return my_dfa
+
+    def test_identify(self):
+
+        symm = SymmBreak.BFS
+
+        self.solve_and_check(self.problem1, symm_mode = symm)
+
+        my_dfa = self.solve_and_check(self.problem2, symm_mode = symm)
+        write_dot(my_dfa, os.path.join(self.outdir,"test1.dot"))
+
+        my_dfa = self.solve_and_check(self.problem3, symm_mode = symm)
+        write_dot(my_dfa, os.path.join(self.outdir,"test2.dot"))
+
+        my_dfa = self.solve_and_check(self.problem5, symm_mode = symm)
+        write_dot(my_dfa, os.path.join(self.outdir,"test3.dot"))
     
-    my_dfa = find_dfa(accepting=accepting, rejecting=rejecting)
+    def test_identify_2(self):
+
+        input_sz = 200
+        
+        symm = SymmBreak.NONE
+        start = 0
+        end = 5
+        step = 1
+        problems = {}
+
+        for n in range(start, end, step):
+            problems[n] = self.generate_problem(8, input_sz, bypass=True)
+            # problems[n] = self.generate_problem(n, input_sz, bypass=True)
+
+        print("Running tests for no symmetry breaking...")
+        for n in range(start, end, step):
+
+            t0 = time.time()
+            dfa1 = self.solve_and_check(problems[n], symm_mode=symm)
+            elapsed = time.time() - t0
+            
+            print("solved n = {} with no symmetry-breaking in {:.4g}s".format(n,elapsed))
+            write_dot(dfa1, os.path.join(self.outdir,"longtest_n-{}_symm-{}.dot".format(n,symm)))
+        
+        symm = SymmBreak.CLIQUE
+        print("Running tests for clique-based symmetry breaking...")
+        for n in range(start, end, step):
+
+            t0 = time.time()
+            dfa1 = self.solve_and_check(problems[n], symm_mode=symm)
+            elapsed = time.time() - t0
+            
+            print("solved n = {} with clique-based symmetry-breaking in {:.4g}s".format(n,elapsed))
+            write_dot(dfa1, os.path.join(self.outdir,"longtest_n-{}_symm-{}.dot".format(n,symm)))
+        
+        symm = SymmBreak.BFS
+        print("Running tests for BFS symmetry breaking...")
+        for n in range(start, end, step):
+            
+            t0 = time.time()
+            dfa1 = self.solve_and_check(problems[n], symm_mode=symm)
+            elapsed = time.time() - t0
+            
+            print("solved n = {} with BFS symmetry-breaking in {:.4g}s".format(n,elapsed))
+            write_dot(dfa1, os.path.join(self.outdir,"longtest_n-{}_symm-{}.dot".format(n,symm)))
+
+    def test_identify_repeatedly(self):
+
+        symm = SymmBreak.BFS
+
+        for i in range(50):
+            self.solve_and_check(self.problem1, symm_mode = symm)
+
+        for i in range(50):
```

Comment:
This is just checking determinism right?

DFA is getting an equiv check which we could use. Alternatively, we could check that the SAT solution is always the same model.

Reply:
50 is also probably too much outside of a benchmark. 3 should be sufficient unless your experience says otherwise.